### DISCUSSION TITLE
Tell platform to always request file from my server

### DISCUSSION THREAD

**Discussion by redimongo** [Podcastindex-org/podcast-namespace#425] [Ideas]
So this at the heart of it, goes to Spotify and their practice of storing our podcasts on their servers - they claim it is for faster service and for "better listening experience", which it might be. However unless you're hosting with a "supported provider" Spotify and some other players will clone your media files to their servers. This hinders the ability for creators to insert ads from other ad serving companies or even replace a section of audio in a file due to error or something.

We have had times where we have had to send Spotify support an email saying please fetch new file for X episode. This not only hurts our listeners, but also makes us have to do more support ticket filing than we should.

I would then suggest that a new tag be developed and supported that stats cloning allowed or not.

```
<podcast:clone>no</podcast:clone>
```
I understand that for most people this won't affect you. But for media companies that host podcasts on their own platform and offer it on Spotify it is something that bugs us.

**Comment by jamescridland** [comment]
> However unless you're hosting with a "supported provider" Spotify and some other players will clone your media files to their servers.

Only Spotify does this. You can stop it from doing it if you go into Spotify for Podcasters for your podcast, and change the "hosting provider" to a company that has pass-through (i.e. that doesn't cache your podcast in its system). They don't check the RSS feed is actually hosted there; but it stops Spotify from caching audio files. Here's a screenshot of the bit I mean.

<img width="627" alt="Screen Shot 2022-12-30 at 3 14 16 pm" src="https://user-images.githubusercontent.com/231941/210037049-d78bb2a7-4932-4a5f-bd08-b7203277114f.png">

> We have had times where we have had to send Spotify support an email saying please fetch new file for X episode.

Change the filename and it'll fetch the new episode audio automatically. That even works if you just change a GET value - `myaudiofile.mp3?version=30dec`

Hope that helps.

**Reply by redimongo** [reply]
Thanks, I'll have to give that a go.

**Comment by daveajones** [comment]
This still might not be a bad idea to generally signal a “nocache” desire.

**Comment by PofMagicfingers** [comment]
It won't be enforced by Spotify. Pass-through or not depends on hosting
provider selection  because hosting provider make special agreements and
comply with specific guidelines for security and such between Spotify and
the hosting provider.

I really doubt Spotify will some day follow such a tag if it exists, as
it's more a "contractual" concern than a technical concern.

Also if you switch the hosting provider on Spotify dashboard you most
likely break the link between Spotify and your hosting provider if you have
one. That means they would not be able to import and include Spotify plays
into your statistics.

The timestamp or checksum in the url is the way to go, usually it's a
checksum of the file or its modification time. But do not try to make it
different at each request, as this would probably break the podcast itself
on many platforms (unstable urls might make your show blacklisted or at
least "frozen")

**Reply by redimongo** [reply]
I would back a nocache tag. We host 68 podcasts and about to launch our system to the public, let's bring this to reality.

**Reply by PofMagicfingers** [reply]
OK sure. But to my knowledge absolutely all platforms and podcast client apps are already "no cache" for enclosures. Only Spotify does copy podcast files to their server. Unless you convince them to follow, i'm not really sure it's worth adding a tag only supported/accepted by people already doing what the tag tries to achieve. 😬 😅

**Reply by redimongo** [reply]
I think the reason we would do this is to make it 100% clear that the owner of the podcast does not give permission for a service to store their files on their server. 

Spotify has technology that also removes audio advertising from podcasts. This can and does end up hurting the podcaster. 

At the end of the day if musicians get paid for their music when played on Spotify then so should podcasters. Podcasting is a creative art form. 

Spotify is benefiting from the millions of podcasters who put their hard work and money into making a podcast. 

It's only fair that if Spotify wants to put ads around podcasts (like they do) and also before podcasts (like they do) then the podcaster should be getting paid. 

Now the only thing we can do is either bring this to people's attention, or come up with a standard that is a fair middle ground that allows us full control over the audio supplied to the end user.

**Reply by jamescridland** [reply]
@redimongo Quite a collection of accusations here.

> I think the reason we would do this is to make it 100% clear that the owner of the podcast does not give permission for a service to store their files on their server.

Spotify only lists a podcast if you, the podcast owner, allows them to. If you don't want to be on Spotify, don't submit your show to Spotify.

> Spotify has technology that also removes audio advertising from podcasts. This can and does end up hurting the podcaster.

I've written about this space since 2017, and nobody has ever made this claim before. Please could you supply any detail?

> It's only fair that if Spotify wants to put ads around podcasts (like they do) and also before podcasts (like they do) then the podcaster should be getting paid.

That's certainly an argument. I'd suggest, though, that Spotify gives podcasts free carriage on their platform, and free stats. They don't interfere with the audio in any way, and don't ask for a revenue share from the ads in the podcasts it carries. And, as stated above, you're entirely free not to be in Spotify if you don't want to be.

**Reply by redimongo** [reply]
Hi @jamescridland 
Yes, as per the license agreement when uploading a podcast to Spotify:

https://www.spotify.com/us/legal/spotify-for-podcasters-platform-terms/
` Accordingly, you hereby grant Spotify a non-exclusive, transferable, sub-licensable, royalty-free, fully paid, worldwide license to: (a) download, digitally transcode, modify, make copies of, create derivative works from (including the right to transcribe, e.g. for the purpose of optimizing search functionalities, improving personalization, simplifying content review and/or displaying on the Spotify Platform, but not including the development and production of derivative audio or video works or the exploitation of customary ancillary rights, e.g. merchandise, live events, etc.), `

The keyword there is modify.

As for ads around podcasts I link to this article from TechCrunch. We don't get paid for these adverts. 
https://techcrunch.com/2022/01/06/spotify-introduces-a-new-ad-format-for-podcasts-that-puts-clickable-cards-inside-shows/

> That's certainly an argument. I'd suggest, though, that Spotify gives podcasts free carriage on their platform, and free stats. They don't interfere with the audio in any way, and don't ask for a revenue share from the ads in the podcasts it carries. And, as stated above, you're entirely free not to be in Spotify if you don't want to be.

While you might be right, the issue is if Spotify wanted to they could/can modify a podcast to remove paid commercial agreements as inline with their terms of service and the license agreement that we as creators agree too. 

I am a firm believer that if a site or service is making money by playing an ad before or during your podcast you as a creator should be paid a percentage.

**Reply by jamescridland** [reply]
We'll probably not agree on this, but a licence saying they can "modify" doesn't mean that they can strip out the ads, nor does it even say it has any technology to do so. Indeed, the licence goes on to say **it specifically won't do that** - since the licence is "not including the development and production of derivative audio or video works".

The part of that licence is specifically around transcriptions and topic tagging, not making different versions - far less, stripping the ads out.

Your allegation was: "Spotify has technology that also removes audio advertising from podcasts. This can and does end up hurting the podcaster."  - but it's clear that no, it _doesn't_ have technology that removes audio advertising - there's no evidence of that - and the allegation that this "can and does end up hurting the podcaster" is nonsense, since it isn't doing it.

Let's have a sensible, grown-up conversation here, not a conspiracy theory fuelled fantasy.

> I am a firm believer that if a site or service is making money by playing an ad before or during your podcast you as a creator should be paid a percentage.

Excellent. So, don't submit your show to Spotify, and they can't "earn from it". On the other hand, you lose free carriage on one of the largest podcast platforms on the planet. :shrug:

**Reply by redimongo** [reply]
If you're asking is it possible to block ads in podcasts yes - https://www.adblockpodcast.com
I normally listen to podcast called Crypto News Alerts it normally has about 6 ads in it. This site has managed to remove the ads. Now as for "is Spotify doing this" I remember reading something a few years ago saying that they "could" do this. The technology does exists.

**Reply by jamescridland** [reply]
> If you're asking is it possible to block ads in podcasts

I'm not. I'm asking "Is Spotify doing this, as you allege they are?"

> I remember reading something a few years ago saying that they "could" do this

I remember reading somewhere that trolls exist even in Github.

**Comment by jamescridland** [comment]
My suggestion on how to achieve this is as an optional value to the enclosure tag.

As a creator, I'm happy for this file to be cached at a podcast platform:

```xml
<enclosure url="https://example.com/audio.mp3" type="audio/mpeg" length="1994643" cacheable="yes" />
```

As a creator, I'm signalling that I am not happy for this file to be cached at a podcast platform:

```xml
<enclosure url="https://example.com/audio.mp3" type="audio/mpeg" length="1994643" cacheable="no" />
```

There are many benefits for podcast audio to be cached by a podcast platform: the platform pays for the bandwidth, not you, as one example. Something that allows creators to have control is the important thing.

However, as @PofMagicfingers points out, the only people currently caching audio is Spotify,  and they only cache audio in a few cases. Without Spotify respecting this tag, then there's little point in working on this.

(One note on Spotify's caching: they take a copy of every single show as a backup anyway, and on some devices will play their cached version irrespective of whether they have a passthrough agreement. That's particularly the case with playback on some of the surfaces that Spotify is available on, like seat-back screens in airlines or Spotify Connect hifi units.)

**Reply by redimongo** [reply]
Yes you could do it on an episode level, but would also be good to have it on a podcast wide level if the podcaster wants to.
Also makes it easier for a podcast hosting provider to integrate a checkbox that states "Would you like your podcast cached on Spotify " or something along the lines of that.

While you're correct we can't stop them from doing this, it would be nice to have the community support the podcasters and not be corporate. If we could get Spotify or others to agree to implement this then at the end of the day the TOS that Spotify has becomes more in the control of the publisher and not Spotify.

This would not stop Spotify or others from delisting a podcast or an episode from a podcast for breaching Spotify content guidelines.

**Reply by jamescridland** [reply]
> If we could get Spotify or others to agree to implement this

...then it's worthwhile working up an example of how it would work. Until that point, I'm not very sure it is. There's no real issue here - if you want stats, Spotify will give them to you; if you want dynamic ads, you'll probably be hosting with a company that uses passthrough anyway. I don't see the use-case.
